src/main/scala/com/codacy/plugins/api/results/Pattern.scala

```diff
@@ -35,6 +35,15 @@ object Pattern {
     val subcategory: Option[Subcategory]
     val parameters: Option[Set[Parameter.Specification]]
     val languages: Option[Set[Language]]
+
+    def copy(patternId: Pattern.Id = patternId,
```

Comment:
the only implementation of `Specification` is `SpecificationImpl`, or we do have a foreseeable use case for keeping it as a `trait` or, since we are breaking already, I propose to change `Specification` to be a `case class` (it will have `copy` and companion `apply` automatically created), this will make the code cleaner and remove a possibly un-necessary abstraction level.

Reply:
I think the reason for it to be a trait was this:
https://github.com/codacy/codacy-plugins-api/blob/e51d6790e13f487a39f3f89a176076a50ec2877c/src/main/scala/com/codacy/plugins/api/results/Pattern.scala#L67
The validation of subcategories.
Can you check if that makes sense to you?

Reply:
I refactored Specification to be a `case class`. Now it is allowed to instantiate a `Specification` where Security subcategory is used as subcategory of another Category (for example `CodeStyle`) without exceptions being thrown.